### PR TITLE
Group options by subcommands

### PR DESCRIPTION
**Pull request by SergeyKutsko**
https://github.com/zold-io/zold/issues/328

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#503](https://github.com/zold-io/zold/issues/503) is now in scope, role is `REV`

File: lib/zold/commands/remote.rb [hunk 1]
```diff
@@ -64,59 +64,57 @@ def run(args = [])
...
-        #  subcommand only.
-        o.integer '--max-nodes',
-          "This applies only to the select subcommand. Number of nodes to limit to. Defaults to #{Remotes::MAX_NODES}.",
-          default: Remotes::MAX_NODES
-        o.bool '--help', 'Print instructions'
+          default: false}
+
+    #{o.bool '--help', 'Print instructions'}"
       end
+
```
Comment (emilianodellacasa) [inline lib/zold/commands/remote.rb]: @SergeyKutsko Empty line is a code smell, please remove it

**Comment by SergeyKutsko** [comment]
@emilianodellacasa  Add test and fixed blank line, thx

**Comment by emilianodellacasa** [comment]
@SergeyKutsko Thanks a lot, it's good to me

**Comment by emilianodellacasa** [comment]
@yegor256 we can merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by SergeyKutsko** [comment]
@yegor256 It is not code issue.

Reported bug: https://github.com/zold-io/zold/issues/509

**Comment by SergeyKutsko** [comment]
@0crat wait

**Comment by 0crat** [comment]
> @0crat wait ([here](https://github.com/zold-io/zold/issues/503#issuecomment-434335948))

@SergeyKutsko It's a code review job [#503](https://github.com/zold-io/zold/issues/503), can't put it on hold

**Comment by emilianodellacasa** [comment]
@yegor256 please retry merging

**Comment by yegor256** [comment]
@rultor merge

**Comment by SergeyKutsko** [comment]
@yegor256 updated

**Comment by SergeyKutsko** [comment]
@rultor merge

**Comment by SergeyKutsko** [comment]
@yegor256 Updated acording feedback

File: test/commands/test_remote.rb [hunk 2]
```diff
@@ -203,4 +203,10 @@ def test_select_respects_max_nodes_option
...
...e node\n    \e[32mremote elect\e[0m\n      Pick a random remote node as a target for a bonus awarding\n      Available options:\n        --ignore-score-weakness  Don't complain when their score is too weak\n        --ignore-score-value  Don't complain when their score is too small\n        --min-score  The minimum score required for winning the election (default: 8)\n        --max-winners  The maximum amount of election winners the election (default: 1)\n    \e[32mremote trim\e[0m\n      Remove the least reliable nodes\n      Available options:\n        --tolerate  Maximum level of errors we are able to tolerate\n    \e[32mremote select [options]\e[0m\n      Select the strongest n nodes.\n      Available options:\n        --max-nodes  Number of nodes to limit to. Defaults to 16.\n    \e[32mremote update\e[0m\n      Check each registered remote node for availability\n      Available options:\n        --ignore-score-weakness Don't complain when their score is too weak\n        --reboot  Exit if any node reports version higher than we have\n        --depth  The amount of update cycles to run, in order to fetch as many nodes as possible (default: 2)\n    --help  Print instructions\n"))
```
Comment (yegor256) [inline test/commands/test_remote.rb]: @SergeyKutsko this anti pattern is called Nitpicker: https://habr.com/post/43761/

**Comment by yegor256** [comment]
@SergeyKutsko see above

**Comment by SergeyKutsko** [comment]
@yegor256 fixed

**Comment by emilianodellacasa** [comment]
@SergeyKutsko you should resolve conflicts before trying to merge again

**Comment by SergeyKutsko** [comment]
@emilianodellacasa @yegor256  Resolved Conflicts. Lets merge.

**Comment by SergeyKutsko** [comment]
@rultor merge

**Comment by SergeyKutsko** [comment]
@yegor256 Can we merge? if can't why?

**Comment by SergeyKutsko** [comment]
@yegor256 can we merge. I can't resolve conflicts again and again.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (48 days), architects (@yegor256) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#503](https://github.com/zold-io/zold/issues/503) is now out of scope

**Comment by ypshenychka** [comment]
@emilianodellacasa According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review, one of which is cosmetic.
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by emilianodellacasa** [comment]
@ypshenychka Confirmed

**Comment by ypshenychka** [comment]
@emilianodellacasa thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request closed without merging**